mlflow/pyfunc/__init__.py

```diff
@@ -2037,6 +2037,11 @@ def predict(model_input: List[str]) -> List[str]:
             # output is not coercable to ChatResponse
             messages = [ChatMessage(**m) for m in input_example["messages"]]
             params = ChatParams(**{k: v for k, v in input_example.items() if k != "messages"})
+
+            # call load_context() first, as predict may depend on it
+            _logger.info("Predicting on input example to validate output")
+            context = PythonModelContext(artifacts, model_config)
+            python_model.load_context(context)
             output = python_model.predict(None, messages, params)
```

Comment:
Should we also pass the context to prediction?

Reply:
oh yes that makes sense!